indico/modules/events/papers/operations.py

```diff
@@ -81,3 +86,24 @@ def close_cfp(event):
     event.cfp.close()
     logger.info("Call for papers for %r closed by %r", event, session.user)
     event.log(EventLogRealm.management, EventLogKind.negative, 'Papers', 'Call for papers closed', session.user)
+
+
+@no_autoflush
+def judge_paper(contribution, contrib_data, judgment, judge, send_notifications=False):
+
+    if judgment == PaperAction.accept:
+        contribution.paper_last_revision.state = PaperRevisionState.accepted
+    elif judgment == PaperAction.reject:
+        contribution.paper_last_revision.state = PaperRevisionState.rejected
+    elif judgment == PaperAction.to_be_corrected:
+        contribution.paper_last_revision.state = PaperRevisionState.to_be_corrected
+    if contrib_data['judgment_comment']:
+        contribution.paper_last_revision.judgment_comment = contrib_data['judgment_comment']
+    db.session.flush()
+    log_data = {'New state': orig_string(judgment.title), 'Sent notifications': send_notifications}
+    if send_notifications:
+        pass  # TODO: manage notifications
+    logger.info('Paper %s changed state by %r to %s', contribution.title, judge,
```

Comment:
`Paper %r ..` and use `contribution` instead of `contribution.title`.
This is for the logfile, having the repr is *much* more useful there than just the title